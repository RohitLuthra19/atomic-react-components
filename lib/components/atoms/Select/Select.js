// @flow
import React from 'react';
import type { Node } from 'react';
import styled from 'styled-components';
import styles from './Select.style';

type Props = {
  id: string,
  className ?: string,
  name: string,
  disabled ?: string | boolean,
  options: any,
  selectedOption ?: string,
  placeholder ?: Node,
};

const Select = ({
  id,
  className,
  disabled,
  placeholder,
  options,
  selectedOption,
  name,
  ...others
}: Props): Node => (
  <select
    id={id}
    name={name}
    className={className}
    value={selectedOption}
    disabled={disabled}
    {...others}
  >
    {placeholder && <option value="">{placeholder}</option>}
    {options.map(opt => (
      <option
        key={opt}
        value={opt}
      >
        {opt}
      </option>
    ))}
  </select>
);

Select.defaultProps = {
  disabled: false,
  selectedOption: '',
  placeholder: '',
  className: '',
};

export default styled(Select)`${styles}`;
export { Select as SelectVanilla };
