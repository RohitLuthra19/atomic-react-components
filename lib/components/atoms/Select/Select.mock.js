import { action } from '@storybook/addon-actions';

const selectDefault = {
  name: 'select',
  options: ['Opt 1', 'Opt 2'],
  onSelect: action('selected'),
};

const selectDisabled = {
  name: 'select',
  options: ['Opt 1', 'Opt 2'],
  disabled: true,
};

export default selectDefault;
export { selectDefault, selectDisabled };
