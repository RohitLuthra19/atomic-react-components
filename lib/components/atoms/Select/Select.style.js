import { css } from 'styled-components';

const styles = css`
    height: auto;
    background: ${props => (props.theme.fieldBackgroundColor)};
    border: 1px solid ${props => (props.theme.fieldBorderColor)};
    border-radius: 0.2rem;
    padding: 0.6rem 1rem;
    color: ${props => (props.theme.fieldTextColor)};


    /* Disabled*/
    ${props => (props.disabled ? `
      background: ${props.theme.fieldBackgroundDisabledColor};
      border-color: ${props.theme.fieldBorderDisabledColor};
    ` : '')};

    ${props => (props.inheritedStyles ? props.inheritedStyles : '')};
`;

export default styles;
