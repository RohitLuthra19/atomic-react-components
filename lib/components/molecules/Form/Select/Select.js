// @flow
import React from 'react';
import type { Node } from 'react';
import { Field, ErrorMessage } from 'formik';
import styled from 'styled-components';
import Label from '../../../atoms/Label';
import AtomSelect from '../../../atoms/Select';
import FieldError from '../../../atoms/FieldError';
import styles from './Select.style';


type Props = {
  id: string,
  className: string,
  name: string,
  label: Node,
  labelProps ?: Object,
  fieldProps ?: Object,
};

const Select = ({
  id,
  className,
  name,
  label,
  labelProps,
  fieldProps,
  ...others
}: Props): Node => (
  <div className={className}>
    <Label htmlFor={id} {...labelProps}>{label}</Label>
    <Field
      name={name}
      component="select"
      render={({ field }) => <AtomSelect {...others} {...field} />}
      {...fieldProps}
    />
    <ErrorMessage name={name} component={FieldError} />
  </div>
);

Select.defaultProps = {
  labelProps: {},
  fieldProps: {},
};

export default styled(Select)`${styles}`;
export { Select as SelectVanilla };
