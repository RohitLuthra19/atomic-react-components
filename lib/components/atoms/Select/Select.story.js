// @flow
import React from 'react';
import { storiesOf } from '@storybook/react';
import { selectDefault } from './Select.mock';

// Import Vanilla Component to display markup and props
import Select, { SelectVanilla } from '.';

// $FlowFixMe eslint-disable-next-line extra-rules/no-commented-out-code
storiesOf('Atoms', module)
  .addParameters({ jest: ['Select', 'SelectVanilla'] })
  .add('Select Knobs', () => (
    <SelectVanilla id="userType" name="username" options={['Option 1', 'Option 2']} />
  ))
  .add('Select', () => <Select {...selectDefault} />);

// eslint-disable-next-line prettier/prettier
/* .addWithChapters('Select', {
    chapters: [
      {
        sections: [
          {
            sectionFn: () => (
              <SelectVanilla id="userType" name="username" options={['Option 1', 'Option 2']} />
            ),
            options: {
              showSource: true,
              allowSourceToggling: true,
              showPropTables: true,
              allowPropTablesToggling: true,
            },
          },
        ],
      },
      {
        title: 'Select Input',
        sections: [
          {
            title: 'Default',
            sectionFn: () => <Select {...selectDefault} />,
          },
          {
            title: 'Disabled',
            sectionFn: () => <Select {...selectDisabled} />,
          },
          {
            title: 'With custom element locator',
            sectionFn: () => <Select {...selectElementLocator} />,
          },
        ],
      },
    ],
  });
 */
