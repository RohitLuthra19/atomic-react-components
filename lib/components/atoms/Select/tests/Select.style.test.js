import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import { SelectVanilla } from '../Select';
// eslint-disable-next-line import/no-unresolved
import Theme from '../../../../styles/theme';

describe('Select Styled component', () => {
  const theme = { ...Theme };

  test('should render correctly', () => {
    const input = shallow(
      <SelectVanilla theme={theme} name="userType" options={['Option 1', 'Option 2']} />
    );
    expect(input).toMatchSnapshot();
  });
});
