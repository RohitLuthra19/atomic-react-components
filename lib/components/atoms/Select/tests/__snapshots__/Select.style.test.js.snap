// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Styled component should render correctly 1`] = `
<select
  className=""
  disabled={false}
  name="userType"
  theme={
    Object {
      "backgroundColorPrimary": "#fff",
      "backgroundColorSecondary": "#eaf2f6",
      "backgroundColorSuccess": "#2e993a",
      "borderColorBoldest": "#c00",
      "borderColorSecondary": "#d9d9d9",
      "brandPrimaryColor": "#006193",
      "buttonBackgroundColorPrimary": "#e57725",
      "buttonBackgroundColorSecondary": "#016fad",
      "buttonColorPrimary": "#ffffff",
      "buttonColorSecondary": "#ffffff",
      "colorOutline": "#999",
      "containerShadowPrimary": "rgba(0,0,0,.2)",
      "fieldBackgroundColor": "#ffffff",
      "fieldBackgroundDisabledColor": "#f3f3f3",
      "fieldBorderColor": "#bfbfbf",
      "fieldBorderDisabledColor": "#bfdfdf",
      "fieldBorderErrorColor": "#c00",
      "fieldBorderFocusColor": "#006193",
      "fieldTextColor": "#222",
      "fontBold": "700",
      "fontBook": "400",
      "fontFamilyDefault": "Arial, Helvetica, sans-serif",
      "fontFamilyIcon": "",
      "fontFamilyPrimary": "Arial, Helvetica, sans-serif",
      "fontFamilySecondary": "Arial, Helvetica, sans-serif",
      "fontMedium": "500",
      "fontNormal": "normal",
      "fontSemiBold": "600",
      "fontSizeBase": "16px",
      "fontSizeButtonPrimary": "24px",
      "fontSizeButtonSecondary": "16px",
      "fontSizeDefault": "12px",
      "fontSizeHighlight": "18px",
      "letterSpacing": "0px",
      "lineHeightBase": "1.5",
      "linkColorPrimary": "#008CBA",
      "linkColorSecondary": "#016fad",
      "maxWidthPageContent": "80rem",
      "textColorBrandPrimary": "#006193",
      "textColorError": "#c00",
      "textColorPlaceholder": "#dfdfdf",
      "textColorPrimary": "#222",
      "textColorSecondary": "#262626",
      "textColorSecondaryLight": "#666",
      "textColorSuccess": "#2e993a",
      "zIndex": Object {
        "zLoader": 1000,
        "zNativeTickSelectIcon": 50,
        "zOverlay": 900,
        "zSticky": 700,
      },
    }
  }
  value=""
>
  <option
    key="Option 1"
    value="Option 1"
  >
    Option 1
  </option>
  <option
    key="Option 2"
    value="Option 2"
  >
    Option 2
  </option>
</select>
`;
